scripts/rust-analyze.sh

```diff
@@ -4,27 +4,54 @@ set -x # Show commands
 set -eu # Errors/undefined vars are fatal
 set -o pipefail # Check all commands in a pipeline
 
-if [ $# -ne 2 ]
+if [ $# -lt 2 ]
 then
-    echo "Usage: rust-analyze.sh config-file.json tree_name"
+    echo "Usage: rust-analyze.sh config-file.json tree_name [platform-filter]"
     exit 1
 fi
 
 CONFIG_FILE=$(realpath $1)
 TREE_NAME=$2
 
+# Our current rust-indexer implementation can get confused if it's given the
+# same crate but for different platforms at the same time.  So we provide an
+# optional mechanism that takes a platform name per mozsearch-mozilla
+# convention and:
+# - filters the `find` invocations to only pick up output for that platform.
+# - puts the output in `analysis-linux64` for example instead of just
+#   `analysis`.
+PLATFORM=${3:-}
+
+# Figure out the name of the platform specific dir used for the rust stuff.
+declare -A RUST_PLAT_DIRS
+RUST_PLAT_DIRS["linux64"]="x86_64-unknown-linux-gnu"
+RUST_PLAT_DIRS["macosx64"]="x86_64-apple-darwin"
+RUST_PLAT_DIRS["win64"]="x86_64-pc-windows-msvc"
+RUST_PLAT_DIRS["android-armv7"]="thumbv7neon-linux-androideabi"
+
+# Hacky mechanism where we take a rust objdir value like x86_64-unknown-linux-gnu
+# to filter the rust analysis to just that.
+PLATFORM_SAVE_FILTER=${PLATFORM:+${RUST_PLAT_DIRS[$PLATFORM]}/debug/deps/}
+PLATFORM_FILTER=${PLATFORM:+${RUST_PLAT_DIRS[$PLATFORM]}/}
+PLATFORM_SUFFIX=${PLATFORM:+-$PLATFORM}
+
 if [ -d "$OBJDIR" ]; then
-  ANALYSIS_DIRS="$(find $OBJDIR -type d -name save-analysis)"
+  # Bail if the build step already performed rust analysis.
+  if [ -f "$OBJDIR/rust-analyzed" ]; then
```

Comment:
This check/bailout seems like it would be better placed around the invocation of this script in mkindex.sh.

Reply:
I am going to temporarily take the punt because my fluency with bash/test conditionals is such that I wouldn't trust the change without running a new try run, especially since it requires effectively inverting the check.  I'll plan to try and make the change and include it in a future try run for future review.